src/components/AvatarWithImagePicker.js

```diff
@@ -116,166 +116,118 @@ const defaultProps = {
     originalFileName: '',
 };
 
-class AvatarWithImagePicker extends React.Component {
-    constructor(props) {
-        super(props);
-        this.animation = new SpinningIndicatorAnimation();
-        this.setError = this.setError.bind(this);
-        this.isValidSize = this.isValidSize.bind(this);
-        this.showAvatarCropModal = this.showAvatarCropModal.bind(this);
-        this.hideAvatarCropModal = this.hideAvatarCropModal.bind(this);
-        this.state = {
-            isMenuVisible: false,
-            validationError: null,
-            phraseParam: {},
-            isAvatarCropModalOpen: false,
-            imageName: '',
-            imageUri: '',
-            imageType: '',
+function AvatarWithImagePicker(props) {
+    const animation = new SpinningIndicatorAnimation();
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [validationError, setValidationError] = useState(null);
+    const [phraseParam, setPhraseParam] = useState({});
+    const [isAvatarCropModalOpen, setIsAvatarCropModalOpen] = useState(false);
+    const [imageName, setImageName] = useState('');
+    const [imageUri, setImageUri] = useState('');
+    const [imageType, setImageType] = useState('');
+    const anchorRef = useRef();
+
+    useEffect(() => {
+        if (props.isUploading) {
```

Comment:
I'm seeing the start, but not seeing the `animation.stop`, the stop from the return only gets executed when the Component is unmounted, but seems that we should stop the animation when the `isUploading` change

Reply:
it is just beneath the line you commented

`    useEffect(() => {
        if (isUploading) {
            animation.start();
        }
        
        return () => {
            animation.stop();
        };
    }, [isUploading]);`